src/protagonist/API/Features/DeliveryChannels/AvChannelPolicyOptionsRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using DLCS.Core.Caching;
+using DLCS.Model.DeliveryChannels;
+using DLCS.Repository.Messaging;
+using LazyCache;
+using Microsoft.Extensions.Options;
+
+namespace API.Features.DeliveryChannels;
+
+public class AvChannelPolicyOptionsRepository : IAvChannelPolicyOptionsRepository
+{
+    private readonly IAppCache appCache;
+    private readonly CacheSettings cacheSettings;
+    private readonly IEngineClient engineClient;
+
+    public AvChannelPolicyOptionsRepository(IAppCache appCache, IOptions<CacheSettings> cacheOptions, IEngineClient engineClient)
+    {
+        this.appCache = appCache;
+        cacheSettings = cacheOptions.Value;
+        this.engineClient = engineClient;
+    }
+
+    public async Task<IReadOnlyCollection<string>?> RetrieveAvChannelPolicyOptions()
+    {
+        const string key = "avChannelPolicyOptions";
+        
+        return await appCache.GetOrAdd(key, () => engineClient.GetAllowedAvPolicyOptions(),
```

Comment:
If we receive a `null` or `empty` value that could signify an issue, either config or service down. If that's the cause would it be worth caching a null object, or caching for a shorter duration? By caching for `Long` if the Engine was down and came back up it would take a while for that to be reflected - or would need the cache to be flushed / service restarted to pickup.

Reply:
Following on from

>would it be worth caching a null object

I don't think `null` values are cached so it would always be a cache miss in that instance.